src/osemgrep/targeting/Glob_matcher.mli

```diff
@@ -0,0 +1,90 @@
+(*
+   AST and matching of a glob pattern against a path.
+   This is purely syntaxic: the file system is not accessed.
+*)
+
+(* The location of a pattern, for logging and troubleshooting. *)
+type loc = {
+  (* File name or other source location name useful to a human reader
+     in error messages. *)
+  source_name : string;
+  (* Line number, starting from 1. *)
+  line_number : int;
+  line_contents : string;
+}
+
+(* Create a location from a pattern string rather than a location in a file. *)
+val string_loc : ?source_name:string -> string -> loc
+val show_loc : loc -> string
+
+type char_class_range = Class_char of char | Range of char * char
```

Comment:
It's two syntaxes for the same thing, semantically. It may or may not be compiled into the same thing later (we don't care because who writes `h[e]llo` anyway?).